core/common/src/main/java/alluxio/worker/block/io/BlockReader.java

```diff
@@ -46,4 +48,18 @@
    * @return channel
    */
   ReadableByteChannel getChannel();
+
+  /**
+   * Transfers data from this reader to the buffer.
+   *
+   * @param buf the byte buffer
+   * @return the number of bytes transferred, -1 if the end of the block is reached
+   * @throws IOException if any I/O errors occur
+   */
+  int transferTo(ByteBuf buf) throws IOException;
```

Comment:
Both transferFrom and transferTo need to use netty API.

For example, transferTo is used to fill in the netty ByteBuf which is later used in netty API. 
the call stack:

DataBuffer buffer = getDataBuffer();  // calls transferTo
channel.writeAndFlush(createOkResponse(buffer));  // without netty buffer, a copy has to be made internally because we cannot use NettyDataBufferV2.

Moreover, only ByteBuf have readBytes(OutputStream), writeBytes(InputStream) APIs